am-lib/src/test/java/uk/gov/hmcts/reform/amlib/internal/utils/SecurityClassificationTest.java

```diff
@@ -42,4 +54,24 @@ void whenRoleSecurityClassificationIsHighEnoughResourceShouldBeVisible(boolean e
             Arguments.of(false, NONE, RESTRICTED)
         );
     }
+
+    private static Stream<FromValueArguments> fromValueArguments() {
+        return Stream.of(
+            new FromValueArguments(NONE, NONE),
+            new FromValueArguments(PUBLIC, NONE, PUBLIC),
+            new FromValueArguments(PRIVATE, NONE, PUBLIC, PRIVATE),
+            new FromValueArguments(RESTRICTED, NONE, PUBLIC, PRIVATE, RESTRICTED)
+        );
+    }
+
+    private static class FromValueArguments {
+        private final SecurityClassification roleSecurityClassification;
+        private final SecurityClassification[] securityClassifications;
+
+        private FromValueArguments(SecurityClassification roleSecurityClassification,
+                                   SecurityClassification... securityClassifications) {
```

Comment:
Maybe calling it `expectedSecurityClassifications` would make it bit more readable. I am generally not a big fan of such classes that holds both input and expected output but since you wanted to use varargs you had no choice.

Alternatively you could consider `Arguments.of` as below:

```
Arguments.of(NONE, ImmutableSet.of(NONE)),
Arguments.of(PUBLIC, ImmutableSet.of(NONE, PUBLIC)),
Arguments.of(PRIVATE, ImmutableSet.of(NONE, PUBLIC, PRIVATE)),
Arguments.of(RESTRICTED, ImmutableSet.of(NONE, PUBLIC, PRIVATE, RESTRICTED))
```

In my mind it is bit more readable as you segregate more what is the input and output especially that type for both is same.

Reply:
I prefer the Arguments.of() way. Less code 👍